main.tf

```diff
@@ -227,11 +215,16 @@ data "aws_iam_policy_document" "bastion_host_policy_document" {
 
   statement {
     actions = [
-
-      "kms:Encrypt",
-      "kms:Decrypt"
+      "s3:GetObject",
+      "s3:PutObject"
     ]
-    resources = [aws_kms_key.key.arn]
+    resources = ["${aws_s3_bucket.bucket.arn}/*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-server-side-encryption"
+      values   = ["AES256"]
+    }
```

Comment:
Can you check whether this condition is needed? We have a bunch of S3 bucket creations in our main Terraform repository.

Reply:
For this condition it depends on which  `server_side_encryption_configuration` are we using. If we are using a `sse_algorithm = "AES256"` then this is the condition.

Reply:
Ah, never mind. I was comparing this to bucket policies. That's probably why we have a difference here. Proceed.